lib/generators/hyrax/work/work_generator.rb

```diff
@@ -73,7 +44,7 @@ def register_work
     lastmatch = nil
     in_root do
       File.open(config).each_line do |line|
-        lastmatch = line if line =~ /^\s*config.register_curation_concern :.*\n$/
+        lastmatch = line if line =~ /config.register_curation_concern :(?!#{file_name})/
```

Comment:
Using negative lookahead in the anchor allows for this method to work for revoke/destroy as well as invoke/generate.